tests/icon-api.js

```diff
@@ -0,0 +1,252 @@
+import test from 'brittle'
+import RAM from 'random-access-memory'
+import Database from 'better-sqlite3'
+import { drizzle } from 'drizzle-orm/better-sqlite3'
+import { migrate } from 'drizzle-orm/better-sqlite3/migrator'
+import { randomBytes } from 'node:crypto'
+
+import IconApi, {
+  kCreate,
+  kGetIcon,
+  kGetIconUrl,
+  kGetBestVariant,
+} from '../src/icon-api.js'
+import { DataType } from '../src/datatype/index.js'
+import { DataStore } from '../src/datastore/index.js'
+import { createCoreManager } from './helpers/core-manager.js'
+import { iconTable } from '../src/schema/project.js'
+import { IndexWriter } from '../src/index-writer/index.js'
+
+const expectedSmallIcon = randomBytes(128)
+const expectedMediumIcon = randomBytes(128)
+const expectedLargeIcon = randomBytes(128)
+const cm = createCoreManager()
+const sqlite = new Database(':memory:')
+const db = drizzle(sqlite)
+migrate(db, {
+  migrationsFolder: new URL('../drizzle/project', import.meta.url).pathname,
+})
+
+const indexWriter = new IndexWriter({
+  tables: [iconTable],
+  sqlite,
+})
+
+const iconDataStore = new DataStore({
+  namespace: 'config',
+  coreManager: cm,
+  storage: () => new RAM(),
+  batch: async (entries) => indexWriter.batch(entries),
+})
+
+const iconDataType = new DataType({
+  dataStore: iconDataStore,
+  table: iconTable,
+  db,
+})
+const projectId = randomBytes(32).toString('hex')
+const iconApi = new IconApi({
+  iconDataStore,
+  iconDataType,
+  projectId,
+})
+
+// eslint-disable-next-line no-unused-vars
+test('icon api - create and get with one variant', async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
```

Comment:
`expectedSmallIcon` is already a buffer
```suggestion
        blob: expectedSmallIcon,
```